### ISSUE TITLE
Failing on 0.4 master

### ISSUE THREAD

**Issue by ViralBShah** [JuliaDB/ODBC.jl#84]
````
julia> using ODBC
Warning: requiring "Dates" did not define a corresponding module.
WARNING: dlopen is deprecated, use Libdl.dlopen instead.
 in depwarn at ./deprecated.jl:44
 in dlopen at deprecated.jl:12
 in anonymous at no file:12
 in include at ./boot.jl:252
 in include_from_node1 at ./loading.jl:134
 in include at ./boot.jl:252
 in include_from_node1 at ./loading.jl:134
 in reload_path at ./loading.jl:158
 in _require at ./loading.jl:70
 in require at ./loading.jl:53
while loading /Users/viral/.julia/v0.4/ODBC/src/ODBC_Types.jl, in expression starting on line 2
WARNING: int16(x) is deprecated, use Int16(x) instead.
 in depwarn at ./deprecated.jl:44
 in int16 at deprecated.jl:31
 in include at ./boot.jl:252
 in include_from_node1 at ./loading.jl:134
 in include at ./boot.jl:252
 in include_from_node1 at ./loading.jl:134
 in reload_path at ./loading.jl:158
 in _require at ./loading.jl:70
 in require at ./loading.jl:53
while loading /Users/viral/.julia/v0.4/ODBC/src/ODBC_Types.jl, in expression starting on line 158

WARNING: deprecated syntax "[a=>b, ...]" at /Users/viral/.julia/v0.4/ODBC/src/ODBC_Types.jl:257.
Use "Dict(a=>b, ...)" instead.

WARNING: deprecated syntax "[a=>b, ...]" at /Users/viral/.julia/v0.4/ODBC/src/ODBC_Types.jl:281.
Use "Dict(a=>b, ...)" instead.

WARNING: deprecated syntax "[a=>b, ...]" at /Users/viral/.julia/v0.4/ODBC/src/ODBC_Types.jl:305.
Use "Dict(a=>b, ...)" instead.

WARNING: deprecated syntax "[a=>b, ...]" at /Users/viral/.julia/v0.4/ODBC/src/ODBC_API.jl:42.
Use "Dict(a=>b, ...)" instead.
WARNING: uint(x) is deprecated, use UInt(x) instead.
 in depwarn at ./deprecated.jl:44
 in uint at deprecated.jl:31
 in include at ./boot.jl:252
 in include_from_node1 at ./loading.jl:134
 in include at ./boot.jl:252
 in include_from_node1 at ./loading.jl:134
 in reload_path at ./loading.jl:158
 in _require at ./loading.jl:70
 in require at ./loading.jl:53
while loading /Users/viral/.julia/v0.4/ODBC/src/ODBC_API.jl, in expression starting on line 159
WARNING: uint16(x) is deprecated, use UInt16(x) instead.
 in depwarn at ./deprecated.jl:44
 in uint16 at deprecated.jl:31
 in include at ./boot.jl:252
 in include_from_node1 at ./loading.jl:134
 in include at ./boot.jl:252
 in include_from_node1 at ./loading.jl:134
 in reload_path at ./loading.jl:158
 in _require at ./loading.jl:70
 in require at ./loading.jl:53
while loading /Users/viral/.julia/v0.4/ODBC/src/ODBC_API.jl, in expression starting on line 347
ERROR: LoadError: TypeError: apply_type: in Array, expected Type{T}, got Tuple{DataType,DataType}
 in include at ./boot.jl:252
 in include_from_node1 at ./loading.jl:134
 in reload_path at ./loading.jl:158
 in _require at ./loading.jl:70
 in require at ./loading.jl:53
while loading /Users/viral/.julia/v0.4/ODBC/src/ODBC.jl, in expression starting on line 16
````

**Comment by ScottPJones** [comment]
It looks like fixes where checked in, but doing `Pkg.add("ODBC") ; using ODBC` gets the same errors on 0.4, it looks like it is picking up the old sources.

**Comment by gbarz** [comment]
Current version of this package still causing error with the latest update of Julia at least on Mac OSX version:    I.e.:  using ODBC

"ERROR:  LoadError:  LoadError:  TypeError:  Array:  in parameter, expected Type{T}, got Tuple{DataType,DataType}"

Additionally, query("INSERT INTO....")  causes ****kernel died*** message in Julia 0.3.11, again at least on Mac OSX.

**Comment by VishalGokhale** [comment]
Getting this error on Windows 8
```
ERROR: LoadError: TypeError: Array: in parameter, expected Type{T}, got Tuple{DataType,DataType}
 in include at boot.jl:261
 in include_from_node1 at loading.jl:304
 in require at loading.jl:243
while loading C:\Users\vishal.vishalgokhale\.julia\v0.4\ODBC\src\ODBC.jl, in expression starting on line 16
```
https://github.com/quinnj/ODBC.jl/issues/84#issuecomment-147235611

**Comment by kafisatz** [comment]
I believe there is an easy fix for this: 
line 21

coltypes::Array{Tuple{String, Int16}} #coltypes::Array{(tring, Int16)}

line 81:
const null_meta = Metadata("", 0, 0, UTF8String[], Array(Tuple{String,Int16},0), Int[], Int16[], Int16[]) #modified

**Comment by ViralBShah** [comment]
Would it be possible to provide a PR?

**Comment by kafisatz** [comment]
Unfortunately I have neither the time nor skills to do so. 
Also for the package to work further changes are needed. the above only fixes the loading of the Pkg.
There are more places where the 0.4 tuple syntax creates errors, such as line 57 in backend.jl 
After fixing that, anther conversion error occurs. Such errors are very tedious to find, as there is no reference to the line where it occurs (at least in windows). That is something julia should definitely improve on!!

**Comment by bmharsha** [comment]
This issue is resolved, ODBC is now working on 0.4
